audb/core/publish.py

```diff
@@ -181,8 +181,8 @@ def _find_media(
         media_archives.add(deps.archive(file))
         deps._drop(file)
 
-    # limit to relevant media
-    db_media_in_root = db_media.intersection(db_root_files)
+    # limit to relevant media and impose order
+    db_media_in_root = sorted(db_media.intersection(db_root_files))
```

Comment:
Sorting `db_media_in_root` will have most likely no influence as it is called inside `audeer.run_tasks()` and I don't think multi-threading will preserve the order.

Is it needed to preserve the order here anyway as you also sort the `media_archives` return value?

Reply:
Yes actually it is needed, I tried removing it and the old problem is back. I think because

Reply:
I tried setting `num_workers` to 2, and it's still working fine if we keep the sorting here, doesn't make much sense, I'll try using more files and more workers and verify

Reply:
So as expected, I tried publishing a database with 20 files, using 10 workers while keeping the first sort in place (because removing it makes it fail in the first place), and now we're back to the same problem. It looks like sorting the archives in the end is not enough but executing `job()` in `_find_media()` is what's responsible for the files orders.

I tried to investigate if maybe `_find_media()` isn't what's responsible in the order but `_put_media()`, by hard-coding the num_workers in the latter to 1, but looked like the order kept changing. However when I did the opposite (hard-code `num_workers=1` in `_find_media()`) the order is now kept again).

Conclusion: 
-  The `job` in `_find_media()` is responsible for the media files orders
- The returned archives aren't responsible for the order and removing the sorting while keeping a sequential execution of the job in `find_media()` doesn't have an effect

Proposal:
- We change the job in `find_media()` to always be sequential (not good will result in latency compared to the current solution)
- We apply sorting in a later stage like @frankenjoe suggested earlier

Reply:
Actually there is a simpler way to do this. These few lines are responsible for the order:
![image](https://user-images.githubusercontent.com/38329838/236230796-3fd047fc-cebf-40c7-9886-9619efb0c89b.png)

Reply:
I've also added the order test to `test_publish()` and removed the extra test as suggested by @frankenjoe 

```python
    # Check if media files are sorted in right order
    assert deps.media == sorted(deps.media)
```

Reply:
Yes, alphabetically sorting should always be fine.